ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -53,11 +89,86 @@ const SnapUIRendererComponent = ({
     // We do this to avoid useless re-renders.
     (oldState, newState) => isEqual(oldState.content, newState.content),
   );
-  const rawContent = interfaceState?.content;
-  const content =
-    rawContent?.type === 'Container' || !rawContent
+
+  const content = useMemo(() => {
+    const rawContent = interfaceState?.content;
+    return rawContent?.type === 'Container' || !rawContent
       ? rawContent
       : Container({ children: rawContent });
+  }, [interfaceState?.content]);
+
+  const requireScroll = useMemo(
+    () => Boolean(content?.props?.children?.[1]?.props?.requireScroll),
```

Comment:
Is this guaranteed to be the correct component?

Reply:
yup, footer can only ever be present inside a container as the last child so it'll always be at that index.